src/System/BpmControl.cpp

```diff
@@ -0,0 +1,33 @@
+#include "HardwareInput.h"
+#include "BpmControl.h"
+
+#define ANALOG_MAX 1023
+#define ANALOG_FREQ 20000
+
+BpmControl::BpmControl (pin_size_t pinNum){
+    _pinNum = pinNum;
+}
+
+void BpmControl::init (){
+    pinMode(_pinNum, OUTPUT);
+    analogWriteRange(ANALOG_MAX);
+    analogWriteFreq(ANALOG_FREQ);
+};
+
+void BpmControl::setControlSignal (int32_t controlSignal){ 
+    if (_active == true){
+        if (controlSignal >= 0 && controlSignal <= 1023){
+            analogWrite (_pinNum, controlSignal);
+        }
```

Comment:
If the control signal is > 1023 or < 0, then the calling code isn't using this API properly.  You should probably print an error message here or make an assertion instead of silently ignoring bad input values. It might help you catch a bug or two in the future.

The [error logger header](https://github.com/supermileage/dynamometer-firmware/blob/dev/src/System/ErrorLogger.h) header has a number of macros which you can use